subsys/net/lib/sockets/sockets.c

```diff
@@ -1134,6 +1134,78 @@ static int sock_get_pkt_src_addr(struct net_pkt *pkt,
 	return ret;
 }
 
+#if defined(CONFIG_NET_OFFLOAD)
+static bool net_pkt_remote_addr_is_unspecified(struct net_pkt *pkt)
+{
+	bool ret = true;
+
+	if (net_pkt_family(pkt) == AF_INET) {
+		ret = net_ipv4_is_addr_unspecified((struct in_addr *)
+								  &pkt->remote);
+	} else if (net_pkt_family(pkt) == AF_INET6) {
+		ret = net_ipv6_is_addr_unspecified((struct in6_addr *)
+								  &pkt->remote);
+	}
+
+	return ret;
+}
+
+static int sock_get_offload_pkt_src_addr(struct net_pkt *pkt,
+					 struct net_context *ctx,
+					 struct sockaddr *addr,
+					 socklen_t addrlen)
+{
+	int ret = 0;
+
+	if (!addr || !pkt) {
+		return -EINVAL;
+	}
+
+	if (!net_pkt_remote_addr_is_unspecified(pkt)) {
+		if (IS_ENABLED(CONFIG_NET_IPV4) &&
+		    net_pkt_family(pkt) == AF_INET) {
+			if (addrlen != sizeof(struct sockaddr_in)) {
```

Comment:
changed from `!=` to `<`

Reply:
I suggested couple of days ago the change from `<` to `!=` as I thought that the buffer length should match because the caller would usually anyway have `sizeof(struct sockaddr_in)` as a length. If the user supplies `sizeof(struct sockaddr_in6)`, and we are have IPv4 socket, it could indicate an error by the user. Anyway, both are ok by me.